src/extensions/cp/spec.lua

```diff
@@ -1,6 +1,161 @@
 --- === cp.spec ===
 ---
---- An asynchronous test suite for Lua.
+--- An synchronous/asynchronous test library for Lua.
+---
+--- This library uses a syntax similar to Ruby RSpec or Mocha.js.
+---
+--- ## Simple Synchronous Test
+---
+--- To create a test, create a new file ending with `_spec.lua`. For example, `simple_spec.lua`:
+---
+--- ```lua
+--- local spec          = require "cp.spec"
+--- local it            = spec.it
+---
+--- return it "always passes"
+--- :doing(function()
+---     assert(true, "This always passes")
+--- end)
+--- ```
+---
+--- It can be run from the Debug Console like so:
+---
+--- ```
+--- cp.spec "simple" ()
+--- ```
+---
+--- It will report something like this:
+---
+--- ```
+--- 2019-10-06 18:13:28: [RESULT] it always passes: passed: 1; failed: 0; aborted: 0; time: 0.0022s
+--- ```
+---
+--- ## Simple Synchronous Failure
+---
+--- If a test fails, it gives a report of where it failed, and if provided, the related message:
+---
+--- ```lua
+--- local spec          = require "cp.spec"
+--- local it            = spec.it
+---
+--- return it "always fails"
+--- :doing(function()
+---     assert(false, "This always fails")
+--- end)
+--- ```
+---
+--- This will result in something like this:
+---
+--- ```
+--- 2019-10-06 21:54:16:   [FAIL] it always fails: [.../simple_spec.lua:6] This always fails
+--- 2019-10-06 21:54:16:
+--- 2019-10-06 21:54:16: [RESULT] it always fails: passed: 0; failed: 1; aborted: 0; time: 0.0370s
+--- ```
+---
+--- You can then check the line that failed and resolve the issue.
+---
+--- ## Simple Asynchronous Test
+---
+--- Performing an asynchronous test is only a little more complicated.
+--- We'll modify our `simple_spec.lua` to use of the [Run.This](cp.spec.Run.This.md) instance available to every test:
+---
+--- ```lua
+--- local spec          = require "cp.spec"
+--- local it            = spec.it
+--- local timer         = require "hs.timer"
+---
+--- return it "always passes"
+--- :doing(function(this)
```

Comment:
I’m confused why the mention of Run.This in description above?

Reply:
Or is the parameter returned by for doing() always “this”?

Reply:
The value passed into a `:doing(this)` function will always be a `Run.This` instance. It probably deserves its own section, since it has a couple of functions, regarding both async tests and getting the values from `where` configurations.